src/rules/require-baseline.js

```diff
@@ -0,0 +1,564 @@
+		 * @returns {void}
+		 **/
+		function checkPropertyValueFunction(child) {
+			const propertyValueLevel = types.get(child.name);
+
+			if (propertyValueLevel < baselineLevel) {
+				context.report({
+					loc: child.loc,
+					messageId: "notBaselineType",
+					data: {
+						type: child.name,
+						availability,
+					},
+				});
+			}
+		}
+
+		return {
+			"Atrule[name=supports]"() {
+				supportsRules.push(new SupportsRule());
+			},
+
+			"Atrule[name=supports] > AtrulePrelude > Condition"(node) {
+				const supportsRule = supportsRules.last();
+
+				for (let i = 0; i < node.children.length; i++) {
+					const conditionChild = node.children[i];
+
+					// if a SupportsDeclaration is preceded by "not" then we don't consider it
+					if (
+						conditionChild.type === "Identifier" &&
+						conditionChild.name === "not"
+					) {
+						i++;
+						continue;
+					}
+
+					// save the supported properties and values for this at-rule
+					if (conditionChild.type === "SupportsDeclaration") {
+						const { declaration } = conditionChild;
+						const property = declaration.property;
+						const supportedProperty =
+							supportsRule.addProperty(property);
+
+						declaration.value.children.forEach(child => {
+							if (child.type === "Identifier") {
+								supportedProperty.addIdentifier(child.name);
+								return;
+							}
+
+							if (child.type === "Function") {
+								supportedProperty.addFunction(child.name);
+							}
+						});
+
+						continue;
+					}
+				}
+			},
+
+			"Rule > Block > Declaration"(node) {
+				const property = node.property;
+
+				// ignore unknown properties - no-invalid-properties already catches this
+				if (!properties.has(property)) {
+					return;
+				}
+
+				/*
+				 * Step 1: Check that the property is in the baseline.
+				 *
+				 * If the property has been tested in a @supports rule, we don't need to
+				 * check it because it won't be applied if the browser doesn't support it.
+				 */
+				if (!supportsRules.hasProperty(property)) {
+					const ruleLevel = properties.get(property);
+
+					if (ruleLevel < baselineLevel) {
+						context.report({
+							loc: {
+								start: node.loc.start,
+								end: {
+									line: node.loc.start.line,
+									column:
+										node.loc.start.column +
+										node.property.length,
+								},
+							},
+							messageId: "notBaselineProperty",
+							data: {
+								property,
+								availability,
+							},
+						});
+
+						/*
+						 * If the property isn't in baseline, then we don't go
+						 * on to check the values. If the property itself isn't
+						 * in baseline then chances are the values aren't too,
+						 * and there's no need to report multiple errors for the
+						 * same property.
+						 */
+						return;
+					}
+				}
+
+				/*
+				 * Step 2: Check that the property values are in the baseline.
+				 */
+				for (const child of node.value.children) {
```

Comment:
This would crash when `Declaration#value` is `Raw`, e.g., with tolerant parsing and the following code:

```js
p {
	margin: .;
}
```